src/rules/preferWhileRule.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * @license
+ * Copyright 2018 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { isForStatement } from "tsutils";
+import * as ts from "typescript";
+import * as Lint from "../index";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "prefer-while",
+        description: "Prefer while loops when instead of a for loop without an initializer and incrementor",
+        optionsDescription: "Not configurable.",
+        options: null,
+        optionExamples: [true],
+        hasFix: true,
```

Comment:
Sorry to harp on this again 😊 but could you please add a `rationale` as well? It's not clear from just the `description` why such a thing is better.

Reply:
No problem 🙂 Done in 84c6d8e